3.10/aql/graphs-all-shortest-paths.md

```diff
@@ -0,0 +1,150 @@
+  [LIMIT offset, count]
+```
+
+- `FOR`: emits the variable **path** which contains one path as an object containing 
+   `vertices`, `edges`, and the `weight` of the path.
+- `IN` `OUTBOUND|INBOUND|ANY`: defines in which direction
+  edges are followed (outgoing, incoming, or both)
+- `ALL_SHORTEST_PATHS`: the keyword to compute All Shortest Paths
+- **startVertex** `TO` **targetVertex** (both string\|object): the two vertices between
+  which the paths will be computed. This can be specified in the form of
+  a ID string or in the form of a document with the attribute `_id`. All other
+  values will lead to a warning and an empty result. If one of the specified
+  documents does not exist, the result is empty as well and there is no warning.
+- `GRAPH` **graphName** (string): the name identifying the named graph. Its vertex and
+  edge collections will be looked up.
+- `LIMIT` (see [LIMIT operation](operations-limit.html), *optional*):
+  the maximum number of paths to return.
+
+### Working with collection sets
+
+```aql
+FOR path
+  IN OUTBOUND|INBOUND|ANY ALL_SHORTEST_PATHS
+  startVertex TO targetVertex
+  edgeCollection1, ..., edgeCollectionN
+  [LIMIT offset, count]
+```
+
+Instead of `GRAPH graphName` you can specify a list of edge collections.
+The involved vertex collections are determined by the edges of the given
+edge collections. 
+
+### Traversing in mixed directions
+
+For all shortest paths with a list of edge collections you can optionally specify the
+direction for some of the edge collections. Say for example you have three edge
+collections *edges1*, *edges2* and *edges3*, where in *edges2* the direction
+has no relevance, but in *edges1* and *edges3* the direction should be taken into
+account. In this case you can use `OUTBOUND` as general search direction and `ANY`
+specifically for *edges2* as follows:
+
+```aql
+FOR vertex IN OUTBOUND ALL_SHORTEST_PATHS
+  startVertex TO targetVertex
+  edges1, ANY edges2, edges3
+```
+
+All collections in the list that do not specify their own direction will use the
+direction defined after `IN` (here: `OUTBOUND`). This allows to use a different
+direction for each collection in your path search.
+
+Examples
+--------
+
+We load an example graph to get a named graph that reflects a social network.
+
+![Social Network](../images/social_graph_2.png)
+
+{% arangoshexample examplevar="examplevar" script="script" result="result" %}
+    @startDocuBlockInline GRAPHKSP_01_create_graph
+    @EXAMPLE_ARANGOSH_OUTPUT{GRAPHKSP_01_create_graph}
+    ~addIgnoreCollection("places");
+    ~addIgnoreCollection("connections");
+    var examples = require("@arangodb/graph-examples/example-graph.js");
+    var graph = examples.loadGraph("allShortestPathsGraph");
+    db.places.toArray();
+    db.connections.toArray();
+    @END_EXAMPLE_ARANGOSH_OUTPUT
+    @endDocuBlock GRAPHKSP_01_create_graph
+{% endarangoshexample %}
+{% include arangoshexample.html id=examplevar script=script result=result %}
```

Comment:
note: we will most likely need to introduce a new example graph to the `@arangodb/graph-examples/example-graph.js` for `ALL_SHORTEST_PATHS` demo purposes